File: rathias/day2_2.py
import re

with open("../inputs/day2_1.txt", "r") as f:
    data = f.read()

ranges = data.split(",")
invalidIds = 0
invalidList = []
for rng in ranges:
	parts = rng.split("-")
	first = parts[0]
	last = parts[1]
	for i in range(int(first), int(last) + 1):
		txt = str(i)
		x = re.search(r"^(\d+)\1+$", txt)
		if x:
			invalidIds += i
			invalidList.append(i)
print(invalidIds)

File: rathias/day2_2.lua
local path = "../inputs/day2_test.txt"
local file = assert(io.open(path, "rb"))
local input = file:read("*a")

local function split(str, sep)
	sep = sep or ","
	local list = {}
	for s in string.gmatch(str, "([^" .. sep .. "]+)") do
		list[#list + 1] = s
	end
	return list
end

local ranges = split(input, ",")
local invalidIds = 0
local invalidList = {}
for i, range in ipairs(ranges) do
	local first, last = table.unpack(split(range, "-"))
	for nb = tonumber(first), tonumber(last) do
		local nbStr = tostring(nb)
		print(nbStr)
		print(string.find(nbStr, "^(%d+)+$"))
		if string.find(nbStr, "^((%d+)+)$") then
			invalidIds = invalidIds + nb
			invalidList[#invalidList + 1] = nb
		end
	end
end

print("===============")
-- print(table.unpack(invalidList))
print(invalidIds)
